service/history/api/startworkflow/api.go

```diff
@@ -371,12 +372,12 @@ func (s *Starter) resolveDuplicateWorkflowID(
 	currentWorkflowConditionFailed *persistence.CurrentWorkflowConditionFailedError,
 ) (*historyservice.StartWorkflowExecutionResponse, error) {
 	workflowID := s.request.StartRequest.WorkflowId
-	currentMutableState, err := s.getMutableState(ctx, currentWorkflowConditionFailed.RunID)
+
+	workflowStartTime, err := s.getWorkflowStartTime(ctx, currentWorkflowConditionFailed.RunID)
```

Comment:
I think "workflow" start time is a bit misleading (since workflow can have multiple runs). `currentExecutionStartTime` and `getCurrentExecutionStartTime` is better.

Reply:
will change to currentWorkflowStartTime in the next rev